src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,233 @@
+        line = in.nextLine();
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        while (!line.equals("bye")) { // Exits the program if input is "bye"
+            String[] words = line.split(" ");
+            if (line.equals("list")) {
+                // List out all the tasks added
+                list(tasks, taskCount);
+            } else if (words[0].equals("unmark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as not done
+                unmarkTask(tasks, taskCount, words);
+            } else if (words[0].equals("mark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as done
+                markTask(tasks, taskCount, words);
+            } else {
+                // Adding a task to the list
+                try {
+                    addTask(line, tasks, taskCount);
+                    taskCount++;
+                } catch (IllegalTaskException e) {
+                    // Ignoring empty command inputs
+                }
+            }
+            line = in.nextLine();
+        }
+
+        // Exiting the program
+        exitMessage();
+    }
+
+    private static void addTask(String line, Task[] tasks, int taskCount) throws IllegalTaskException {
+        if (line.contains("/by")) {
+            // Adding a Deadline
+            try {
+                addDeadline(line, tasks, taskCount);
+            } catch (IllegalDeadlineException e) {
+                deadlineErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else if (line.contains("/from") || line.contains("/to")) {
+            // Adding an Event
+            try {
+                addEvent(line, tasks, taskCount);
+            } catch (IllegalEventException e) {
+                eventErrorMessage();
+                throw new IllegalTaskException();
+            } catch (IndexOutOfBoundsException e) {
+                eventErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else if (!line.isBlank()){
+            // Adding a _Todo_
+            try {
+                addTodo(line, tasks, taskCount);
+            } catch (IllegalTodoException e) {
+                todoErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else {
+            throw new IllegalTaskException();
+        }
+    }
+
+    private static void todoErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description.");
+        borderLine();
+    }
+
+    private static void eventErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description, start time and end time");
+        borderLine();
+    }
+
+    private static void deadlineErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description and deadline.");
```

Comment:
The error messages above (lines 103 onwards) have the same starting few words, you can consider abstracting it out as a constant to be reused.